eclair-core/src/main/scala/fr/acinq/eclair/channel/fund/InteractiveTxFunder.scala

```diff
@@ -298,7 +299,9 @@ private class InteractiveTxFunder(replyTo: ActorRef[InteractiveTxFunder.Response
             confirmations_opt <- if (fundingParams.requireConfirmedInputs.forLocal) wallet.getTxConfirmations(txIn.outPoint.txid) else Future.successful(None)
           } yield {
             if (canUseInput(fundingParams, txIn, previousTx, confirmations_opt.getOrElse(0))) {
-              Right(Input.Local(UInt64(0), previousTx, txIn.outPoint.index, txIn.sequence))
+              // Strip input witnesses to save space (there is a max size on txs due to lightning message limits).
+              val previousTxStripped = previousTx.modify(_.txIn.each.witness).setTo(ScriptWitness.empty)
```

Comment:
This doesn't work, it will have already been filtered out by `canUseInput`.
We should do it when fetching the transaction line 298:

```
// Strip input witnesses to save space (there is a max size on txs due to lightning message limits).
previousTx <- wallet.getTransaction(txIn.outPoint.txid).map(tx => tx.modify(_.txIn.each.witness).setTo(ScriptWitness.empty))
```

Reply:
Fixed in [d5e8821](https://github.com/ACINQ/eclair/pull/2755/commits/d5e8821d47425007c1f37476c6c23cd15bafad4b).

Also added a test in [9d5947f](https://github.com/ACINQ/eclair/pull/2755/commits/9d5947ffb32e49218f9fac55b20d40276652007c) and made sure it was failing before the change.